### PR TITLE
Vectra AI with accounts

### PR DESCRIPTION
**Pull request by Vectra-ODuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Adds Vectra accounts related functions + improves incoming mapper

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [x] Yes
       - Further details: Some mapper attributes have been removed (as duplicates) or renamed
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by idovandijk** [comment]
Thank you @Vectra-ODuda for the contribution. While you're waiting for us to start the review, can you please elaborate on the addition of the "accounts"? What is the improvement you're adding in this PR?
The information will help with the review process.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (YuvHayun) [inline Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: please add a note about this in the RN.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (YuvHayun) [inline Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:744]:
```suggestion
        description = summary.get('description', "")
        dst_ips = summary.get('dst_ips', "")
        dst_ports = summary.get('dst_ports', "")
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @Vectra-ODuda, thx for your contribution
Small comments from my side, you can see them below.
Also, update the CommonType RN as well please.

File: Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json
Comment (idovandijk) [inline Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json:27]: This one is related to the ExtraHop integration, I agree it shouldn't be in the CommonTypes pack.
Please see if there's a different field you can use for Vectra Detection ID or you may create a new field in your pack.

**Review by idovandijk — CHANGES_REQUESTED** [review]
Nice job! I only found 1 thing that we need to change :)

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:744]: Hi @YuvHayun, so you prefer to have an empty string than a None value. I'll change that

File: Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json
Comment (Vectra-ODuda) [reply Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json:27]: Hi @idovandijk , indeed I saw that this field related to ExtraHop but being part of the CommonTypes, I was initialy thinking of updating the description to be more generic but also the .json file name to be usable by everyone, not only ExtraHop as quite confusing.
I'll see want I can use instead.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Hi @YuvHayun , regarding the CommonTypes RN, I've asked the question to @kgal-pan but we were unsure how what we need to do.
I'll make the changes.
With that change, how can I put a strong dependency on the minimum CommonTypes pack version to be mandatory for my Pack version ?
Thx in advance

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py]: Done

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Done regarding the Vectra AI Pack RN

Comment (YuvHayun) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Actually what I meant is, that we should add the new param (accounts fetch query) to the RN.

Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Oh ok, makes sense, let me do it also.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (YuvHayun) [inline Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:745]:
```suggestion
    summary = detection.get('summary')
    data_dict = {}
    if summary:
        if detection.get('category'):
            data_dict['Category'] = detection.get('category')
        if summary.get('description', ""):
            data_dict['description'] = summary.get('description', "")
        if summary.get('dst_ips', ""):
            data_dict['dst_ips'] = summary.get('dst_ips', "")
        if summary.get('dst_ports', ""):
            data_dict['dst_ports'] = summary.get('dst_ports', "")
```
Taking back what I said, sorry!

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (YuvHayun) [inline Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:747]:
```suggestion
    return common_extract_data(detection) | data_dict  # noqa: E203
```

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (YuvHayun) [inline Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:750]:
```suggestion
```

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (YuvHayun) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: So just making sure, everything is alright here? no more questions regarding this comment?

**Review by YuvHayun — CHANGES_REQUESTED** [review]
added fixes, they should also fix your Lint & unit tests problems.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Still the strong dependency question now that I've updated the CommonTypes version, is there a way to do that on my pack_metadata file ?

Comment (YuvHayun) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Not sure I understand your questions, can you please elaborate?

File: Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json
Comment (idovandijk) [reply Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json:27]: Yes, let's not touch the existing field as its name already suggests it's for another integration :)

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Sure, let me rephrase. As I was forced to update the "Common Types" pack to allow some of my incident type to be used with Common Types incident fields, I've added a dependency on this pack on my pack_metadata file.
However, just adding the pack name might not be sufficient if use an older version not having my changes.
So I'm asking if there is a way to say Vectra AI Pack v1.1.0 has a dependency on Common Types (minimum version 3.2.47) that way I'm sure my pack will works as expected or the mapping might failed.

Comment (YuvHayun) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: There's no need to mention that anywhere - the dependencies list is calculated automatically and will include the min version for common types.
When users will upgrade your pack it'll also upgrade Common types as well.

File: Packs/Vectra_AI/pack_metadata.json
Comment (YuvHayun) [inline Packs/Vectra_AI/pack_metadata.json:13]:
```suggestion
```
Due to my dependencies answer on the other comment, you can delete that.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:745]: Hi @YuvHayun , if you're OK I've done that, it's quicker and allow me to see all value into a single dict (with ordered keys) and won't change the others parts. Because `summary` key is always present, so it's mainly for PyPy to be happy.
```
summary = detection.get('summary')
if summary:
    description = summary.get('description')
    dst_ips = summary.get('dst_ips')
    dst_ports = summary.get('dst_ports')
else:
    description = dst_ips = dst_ports = None
```

File: Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json
Comment (Vectra-ODuda) [reply Packs/CommonTypes/IncidentFields/incidentfield-ExtraHop_Detection_ID.json:27]: Hi @idovandijk , so far, the “ExtraHop” usage is only visible under the `description`, neither on the `name` or `id` or `cliName` directly. I’ve put aside the `associatedTypes` as this one can have multiples values for any fields.
This is only visible on the .json file name.
Are you OK if I simply update the description to be more generic ? So it could be used by any other integration.
Also should I change the file name (with a git move) or should I left it intact ?
Bonus question: Should I do the same changes on the two others "ExtraHop" incident fields `Detection Update Time`and `Detection End Time` ?
Thx in advance

File: Packs/Vectra_AI/pack_metadata.json
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/pack_metadata.json]: Done as suggested

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:749]: See suggested committed fix above for consistency/readability if you're ok

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py:752]: See suggested committed fix above for consistency/readability if you're ok

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.yml:85]: Ok, I have removed the dependency as suggested

**Comment by Vectra-ODuda** [comment]
Hi, how can I fix the mapper backward compatibly error message on CircleCI as my pack is on Beta ?
Is there is missing flag on the pack_metadata file ?

> ERROR]: Packs/Vectra_AI/Classifiers/classifier-Vectra_Detect_-_Incoming_Mapper.json: [MP107] - Some incident fields were removed from the mapper, The removed fields: {'Vectra Host': {'Vectra Threat Score', 'Is Active', 'Vectra Certainty Score'}, 'Vectra Detection': {'State', 'occurred'}, 'dbot_classification_incident_type_all': {'Source IP', 'External Sub Category Name', 'Detection URL', 'Alert ID', 'External Start Time', 'External Category Name', 'Incident Link', 'Description', 'External End Time'}, 'Vectra Account': {'Is Active', 'Vectra Threat Score', 'Vectra Certainty Score'}}

Some of them as moved to "type_all" instructions, some have been renamed.
Thx in advance

**Comment by YuvHayun** [comment]
> Hi, how can I fix the mapper backward compatibly error message on CircleCI as my pack is on Beta ? Is there is missing flag on the pack_metadata file ?
> 
> > ERROR]: Packs/Vectra_AI/Classifiers/classifier-Vectra_Detect_-_Incoming_Mapper.json: [MP107] - Some incident fields were removed from the mapper, The removed fields: {'Vectra Host': {'Vectra Threat Score', 'Is Active', 'Vectra Certainty Score'}, 'Vectra Detection': {'State', 'occurred'}, 'dbot_classification_incident_type_all': {'Source IP', 'External Sub Category Name', 'Detection URL', 'Alert ID', 'External Start Time', 'External Category Name', 'Incident Link', 'Description', 'External End Time'}, 'Vectra Account': {'Is Active', 'Vectra Threat Score', 'Vectra Certainty Score'}}
> 
> Some of them as moved to "type_all" instructions, some have been renamed. Thx in advance

If @idovandijk approve this changes, we'll force merge this.

**Comment by idovandijk** [comment]
> > Hi, how can I fix the mapper backward compatibly error message on CircleCI as my pack is on Beta ? Is there is missing flag on the pack_metadata file ?
> > > ERROR]: Packs/Vectra_AI/Classifiers/classifier-Vectra_Detect_-_Incoming_Mapper.json: [MP107] - Some incident fields were removed from the mapper, The removed fields: {'Vectra Host': {'Vectra Threat Score', 'Is Active', 'Vectra Certainty Score'}, 'Vectra Detection': {'State', 'occurred'}, 'dbot_classification_incident_type_all': {'Source IP', 'External Sub Category Name', 'Detection URL', 'Alert ID', 'External Start Time', 'External Category Name', 'Incident Link', 'Description', 'External End Time'}, 'Vectra Account': {'Is Active', 'Vectra Threat Score', 'Vectra Certainty Score'}}
> > 
> > 
> > Some of them as moved to "type_all" instructions, some have been renamed. Thx in advance
> 
> If @idovandijk approve this changes, we'll force merge this.

@Vectra-ODuda can you please declare these changes as breaking backward compatibility in the release notes? After that I will approve the PR

**Comment by Vectra-ODuda** [comment]
> > > Hi, how can I fix the mapper backward compatibly error message on CircleCI as my pack is on Beta ? Is there is missing flag on the pack_metadata file ?
> > > > ERROR]: Packs/Vectra_AI/Classifiers/classifier-Vectra_Detect_-_Incoming_Mapper.json: [MP107] - Some incident fields were removed from the mapper, The removed fields: {'Vectra Host': {'Vectra Threat Score', 'Is Active', 'Vectra Certainty Score'}, 'Vectra Detection': {'State', 'occurred'}, 'dbot_classification_incident_type_all': {'Source IP', 'External Sub Category Name', 'Detection URL', 'Alert ID', 'External Start Time', 'External Category Name', 'Incident Link', 'Description', 'External End Time'}, 'Vectra Account': {'Is Active', 'Vectra Threat Score', 'Vectra Certainty Score'}}
> > > 
> > > 
> > > Some of them as moved to "type_all" instructions, some have been renamed. Thx in advance
> > 
> > 
> > If @idovandijk approve this changes, we'll force merge this.
> 
> @Vectra-ODuda can you please declare these changes as breaking backward compatibility in the release notes? After that I will approve the PR

This is now done. Let me know if it's ok for you.

File: Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/Integrations/VectraDetect/VectraDetect.py]: Thx for your `remove_empty_elements` trick on your committed fix @YuvHayun.
I will use that one more often :)

**Review by idovandijk — APPROVED** [review]
Great job!

**Comment by Vectra-ODuda** [comment]
Hi @idovandijk, I've pushed the modifications regarding the following ExtraHop Incident fields `Detection ID`, `Detection End Time` and `Detection Update Time` to be clean by removing description (as on all generic fields) also with a RN addition regarding the descriptions deletion.
Let me know if it's OK for you

**Comment by Vectra-ODuda** [comment]
> Hi @idovandijk, I've pushed the modifications regarding the following ExtraHop Incident fields `Detection ID`, `Detection End Time` and `Detection Update Time` to be clean by removing description (as on all generic fields) also with a RN addition regarding the descriptions deletion. Let me know if it's OK for you
> 
> [Edit] The ciclici: Run validations is not happy on the "Common Types" pack with this error on the 3 changes `[IF115] - The unsearchable key in indicator and incident fields should be set to true.`however this attribute is set to false on any incident fields of this Pack. Maybe you (Cortex Team) should handle that :)

Hi @idovandijk , @YuvHayun , any news on this one ? this is the only blocker I see for this PR.
Thx in advance

File: Packs/Vectra_AI/ReleaseNotes/1_1_0.md
Comment (YuvHayun) [inline Packs/Vectra_AI/ReleaseNotes/1_1_0.md:2]:
```suggestion
**/!\ Caution: This version breaks backward compatibility due to removed/renamed incident fields when Feching Incidents for ***classifier-Vectra_Detect_-_Incoming_Mapper*** mapper.**  
**Please find more details on the Mapper changes. /!\\**
```

**Comment by YuvHayun** [comment]
Hi @Vectra-ODuda, I added 1 comment to fix the RN warning.
Also, waiting for @idovandijk input about the pr.

**Comment by Vectra-ODuda** [comment]
> Hi @Vectra-ODuda, I added 1 comment to fix the RN warning.
> Also, regarding the unsearchable keys - it's up to you to choose if you want these fields to be searchable or not.

Hi @YuvHayun , regarding the unsearchable key, the warning occurs on the "Common Types" pack validation, not on the Vectra AI one. As this pack is mainly maintained by Palo Alto, I'm not sure why the validator complains about it as I've just removed the `description` attribute and that I've left the `unsearchable` key untouched.

File: Packs/Vectra_AI/ReleaseNotes/1_1_0.md
Comment (Vectra-ODuda) [reply Packs/Vectra_AI/ReleaseNotes/1_1_0.md]: done

**Comment by idovandijk** [comment]
> > Hi @Vectra-ODuda, I added 1 comment to fix the RN warning.
> > Also, regarding the unsearchable keys - it's up to you to choose if you want these fields to be searchable or not.
> 
> Hi @YuvHayun , regarding the unsearchable key, the warning occurs on the "Common Types" pack validation, not on the Vectra AI one. As this pack is mainly maintained by Palo Alto, I'm not sure why the validator complains about it as I've just removed the `description` attribute and that I've left the `unsearchable` key untouched.

Hi @Vectra-ODuda, since the fields failing the build are common ones and you only changed the incident associations, we will force merge this build (there's no issue that you need to fix here)

**Review by YuvHayun — APPROVED** [review]

**Comment by YuvHayun** [comment]
@Vectra-ODuda Merging this pr, we'll take care of the force merge.
Thx for your contribution!

**Pull request merged**
